src/sentry/integrations/jira/integration.py

```diff
@@ -48,6 +58,167 @@ def create_comment(self, issue_id, comment):
     def search_issues(self, query):
         return self.get_client().search_issues(query)
 
+    def make_choices(self, x):
+        return [(y['id'], y['name'] if 'name' in y else y['value']) for y in x] if x else []
+
+    def build_dynamic_field(self, group, field_meta):
+        """
+        Builds a field based on JIRA's meta field information
+        """
+        schema = field_meta['schema']
+
+        # set up some defaults for form fields
+        fieldtype = 'text'
+        fkwargs = {
+            'label': field_meta['name'],
+            'required': field_meta['required'],
+        }
+        # override defaults based on field configuration
+        if (schema['type'] in ['securitylevel', 'priority']
+                or schema.get('custom') == JIRA_CUSTOM_FIELD_TYPES['select']):
+            fieldtype = 'select'
+            fkwargs['choices'] = self.make_choices(field_meta.get('allowedValues'))
+        elif field_meta.get('autoCompleteUrl') and \
+                (schema.get('items') == 'user' or schema['type'] == 'user'):
+            pass
+            # TODO(jess): implement autocomplete for users
```

Comment:
actually implementing this is somewhat involved, so wanted to just do that in a separate PR